packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -121,6 +125,7 @@ class ThemeData {
     primaryTextTheme ??= primaryIsDark ? Typography.white : Typography.black;
     iconTheme ??= isDark ? const IconThemeData(color: Colors.white) : const IconThemeData(color: Colors.black);
     primaryIconTheme ??= primaryIsDark ? const IconThemeData(color: Colors.white) : const IconThemeData(color: Colors.black);
+    platform ??= (Platform.isIOS || Platform.isMacOS) ? TargetPlatform.iOS : TargetPlatform.android;
```

Comment:
and that getter should throw an exception saying that a new enum should be added if it's a new platform